happy/HappyNodeRoute.py

```diff
@@ -488,20 +491,33 @@ def __update_state(self):
             self.removeNodeRoute(self.node_id, self.to)
 
     def run(self):
-        self.__pre_check()
-        if self.route_table is not None:
-            with self.getStateLockManager(lock_id="rt"):
-                if self.add:
-                    self.__add_route()
-                else:
-                    self.__delete_route()
+        # query node's route ip in v4 or v6 format.
+        if not self.add and not self.delete:
+            node_route_prefix = self.getNodeRoutePrefix(self.route_type, self.node_id)
+            node_route_via_id = self.getNodeRouteVia(self.route_type, self.node_id)
+            node_route_addrs = self.getNodeAddressesOnPrefix(node_route_prefix, node_route_via_id)
+
+            emsg = "virtual node: {}, route_type: {}, route ip: {}".format(
+                self.node_id, self.route_type, node_route_addrs)
+            print emsg
         else:
-            if self.add:
-                self.__add_route()
-            else:
-                self.__delete_route()
-        self.__post_check()
-        with self.getStateLockManager():
-            self.__update_state()
-            self.writeState()
+            self.__pre_check()
+            # only configure route for nonTAP device
+            # for TAP device, route will be configured in lwip stack in weave.
+            if not self.IsTapDevice(self.node_id):
+                if self.route_table is not None:
+                    with self.getStateLockManager(lock_id="rt"):
```

Comment:
WHy are we passing in the `lock_id`?   Seems questionable in a parallel execution of happy topologies

Reply:
That was original code, I only moved them under `if not self.IsTapDevice(self.node_id):`
I would suggest to keep it as before, in case break something I haven't learnt.